doc/intro.md

```diff
@@ -1,3 +1,64 @@
+
+## Component
+Component is defined as a *tiny Clojure framework for managing the lifecycle and dependencies of software components with runtime state*. This means the Component can be seen as a dependency injection using immutable data structures. To give more context about it, you can watch Sierra's talk [Components Just Enough Structure](https://youtu.be/13cmHf_kt-Q?si=VribqpkKYOofgAWz).
+
+To handle this entire explanation, imagine that you have a database connection to get. If you open so many connections and don't close them, your database will *break* as you increase the opened connections. By default, in object-oriented programming languages we can manage this as an object in memory - only one - that you pass as your parameters by your methods and use that connection - that's where the component came from: you have instanced objects in memory and you have to pass all of them by parameters to access what that object gives you (in our case of a database example, you can make any database operation with it). 
+
+But the Component *magic* is more than that: is how you infer your dependency injection. To understand how it works, see the example below:
+```clojure
+(defn- build-system-map []
+  (component/system-map
+   :config (config/new-config)
+   :http (http/new-http)
+   :router (router/new-router routes/routes)
+   :database (component/using (database/new-database) [:config])
+   :webserver (component/using (webserver/new-webserver) [:config :http :router :database])))
+```
+> The above example is from [microservice-boilerplate](https://github.com/parenthesin/microservice-boilerplate).
+
+You have some components that are configured: you have a `:config`, `:http`, `:router`, `:database` and `:webserver`. Look at the `:config` component: it only has itself, so this represents that `:config` doesn't have any dependency by default from another component. In another hand, when we look at `:database` we have `(component/using (database/new-database) [:config])`, but what that means? Simple: we have a component that starts with `(database/new-database)` but it depends of another component: `:config`! This means that `:database` will only start when `:config` is properly started! And now look to `:webserver`... It will start only after the initialization of `:config`, `:http`, `:router` and `:database`!
+
+By default, the Component library implements a protocol that you have to refer to when you're building your components (a protocol is very similar to an interface from object-oriented programming) - the protocol name is Lifecycle, and it needs to have two main functions: a start and a stop function. Look at the example below:
+```clojure
+(ns com.example.your-application
+  (:require [com.stuartsierra.component :as component]))
+
+(defrecord Database [host port connection]
+  ;; Implement the Lifecycle protocol
+  component/Lifecycle
```

Comment:
you will explore the possibility of implement other protocols here and how this can benefits in a mock components?

Reply:
Great! I will explore that and its benefits in a mock component.